btk/measure.py

```diff
@@ -1,119 +1,73 @@
+    NOTE: Assumes at least 3 bands are present in the images produced.
 
-        Args:
-            image (np.ndarray): Image (single band) of galaxy to perform measurement
+    Args:
+        batch (dict): Output of DrawBlendsGenerator object's `__next__` method.
+        idx (int): Index number of blend scene in the batch to preform
+            measurement on.
 
-        Returns:
-                centers: x and y coordinates of detected  centroids
-        """
-        # set detection threshold to 5 times std of image
-        threshold = 5 * np.std(image)
-        coordinates = peak_local_max(image, min_distance=2, threshold_abs=threshold)
-        return np.stack((coordinates[:, 1], coordinates[:, 0]), axis=1)
-
-    def get_deblended_images(self, data, index):
-        """Returns scarlet modeled blend and centers for the given blend"""
-        image = np.mean(data["blend_images"][index], axis=2)
-        peaks = self.get_centers(image)
-        return {"deblend_image": None, "peaks": peaks}
-
-
-class SepParams(MeasurementParams):
-    """Class to perform detection and deblending with SEP"""
-
-    def __init__(self):
-        self.catalog = None
-        self.segmentation = None
-
-    def get_centers(self, image):
-        """Return centers detected when object detection and photometry
-        is done on input image with SEP.
-        It also initializes the self.catalog and self.segmentation attributes
-        of the class object.
-        Args:
-            image: Image (single band) of galaxy to perform measurement on.
-        Returns:
-            centers: x and y coordinates of detected  centroids
-        """
-        bkg = sep.Background(image)
-        self.catalog, self.segmentation = sep.extract(
-            image, 1.5, err=bkg.globalrms, segmentation_map=True
-        )
-        centers = np.stack((self.catalog["x"], self.catalog["y"]), axis=1)
-        return centers
+    Returns:
+        dict with the centers of sources detected by SEP detection
+        algorithm.
+    """
 
-    def get_deblended_images(self, data, index):
-        """Performs SEP detection on the band-coadd image and returns the
-        detected peaks.
-        Args:
-            data (dict): Output generated by btk.draw_blends containing blended
-                images, isolated images, observing conditions and blend
-                catalog, for a given batch.
-            index (int): Index number of blend scene in the batch to preform
-                measurement on.
-        Returns:
-            dict with the centers of sources detected by SEP detection
-            algorithm.
-        """
-        image = np.mean(data["blend_images"][index], axis=2)
-        peaks = self.get_centers(image)
-        return {"deblend_image": None, "peaks": peaks}
+    image = np.mean(batch["blend_images"][idx], axis=2)
+    bkg = sep.Background(image)
+    catalog = sep.extract(image, 1.5, err=bkg.globalrms, segmentation_map=False)
```

Comment:
Here we can easily get the segmentation with no additionnal effort, and use it to get some basic deblended images ; I believe it will be useful both as an example and as a way to test the metrics part later. I can do it if you prefer.

Reply:
I just added it, let me know if that makes sense (I had removed it and was planning on adding it back later anyways)

Reply:
This got me thinking, maybe we want to enforce that every algorithm returns a `peak` key because otherwise there is no way of telling how to match the `deblended_images` or `segmentation` to which isolated objects.

Reply:
I think you can match even if you don't have the peaks, using intersection over union or something like that. It is easier to do with the peaks though.
The catalog idea is pretty good ; we should still make sure that people use standardized names but I think it would work great if we want to have support for photometric redshifts or shape measurement for instance.

Reply:
As for SEP, we can even have deblended_images by using the mask from the segmentation to recover the pixels corresponding to each object. I also think the raw segmentation from sep isn't a good format for segmentation in general as you can only have one object corresponding to each pixel ; we should instead have a (stamp_size,stamp_size,n_objects) boolean array.

Reply:
Thanks for reworking the SEP part, I agree with your reasoning. I'll try to add a catalog format with standard names.